File: rest-star/messaging/session-interface/examples/multi-language-demo/receiveOrder.py
import httplib, urlparse

conn = httplib.HTTPConnection("localhost:9095")
conn.request("HEAD", "/queues/jms.queue.orders")
res = conn.getresponse()
consumeLink = res.getheader("msg-consume-next")
print consumeLink
conn.close()

headers = {"Accept-Wait" : "3"}

try:
    while True:
        createParsed = urlparse.urlparse(consumeLink)
        conn = httplib.HTTPConnection(createParsed.netloc)
        print "Waiting..."
        conn.request("POST", createParsed.path, None, headers)
        res = conn.getresponse()
        if res.status == 503:
            print "Timeout on receive, retrying."
            session = res.getheader("msg-session")
            consumeLink = res.getheader("msg-consume-next")
        elif res.status == 200:
            data = res.read()
            print data
            consumeLink = res.getheader("msg-consume-next")
            session = res.getheader("msg-session")
        else:
            raise Exception('failed')
finally:
    if session != None:
        print "deleting hornetq session..."
        createParsed = urlparse.urlparse(session)
        conn = httplib.HTTPConnection(createParsed.netloc)
        conn.request("DELETE", createParsed.path)
        res = conn.getresponse()
        
        
    

File: rest-star/messaging/session-interface/examples/dup-send/src/main/java/ReceiveOrder.java
import org.hornetq.jms.client.HornetQDestination;
import org.hornetq.rest.Jms;
import org.jboss.resteasy.client.ClientRequest;
import org.jboss.resteasy.client.ClientResponse;
import org.jboss.resteasy.spi.Link;

import javax.jms.Connection;
import javax.jms.ConnectionFactory;
import javax.jms.Destination;
import javax.jms.JMSException;
import javax.jms.Message;
import javax.jms.MessageConsumer;
import javax.jms.MessageListener;
import javax.jms.ObjectMessage;
import javax.jms.Session;

/**
 * @author <a href="mailto:bill@burkecentral.com">Bill Burke</a>
 * @version $Revision: 1 $
 */
public class ReceiveOrder
{
   public static void main(String[] args) throws Exception
   {
      System.out.println("here");
      // first get the create URL for the shipping queue
      ClientRequest request = new ClientRequest("http://localhost:9095/queues/jms.queue.orders");
      ClientResponse res = request.head();
      Link pullConsumers = res.getHeaderAsLink("msg-pull-consumers");
      res = pullConsumers.request().post();
      Link consumeNext = res.getHeaderAsLink("msg-consume-next");
      System.out.println("here2");
      while (true)
      {
         System.out.println("Waiting...");
         res = consumeNext.request().header("Accept-Wait", "10").post();
         if (res.getStatus() == 503)
         {
            System.out.println("Timeout...");
            consumeNext = res.getHeaderAsLink("msg-consume-next");
         }
         else if (res.getStatus() == 200)
         {
            Order order = (Order)res.getEntity(Order.class);
            System.out.println(order);
            consumeNext = res.getHeaderAsLink("msg-consume-next");
         }
         else
         {
            throw new RuntimeException("Failure! " + res.getStatus());
         }
      }
   }
}